### PR TITLE
Add unstyled switch

### PR DESCRIPTION
**Pull request by Magnus-G**
image

### CODE REVIEW

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  cursor: pointer;
+  display: inline-block;
+  position: relative;
+
+  input[type="checkbox"] {
+    display: none;
+
+    + .checkbox {
+      @include size($switch-width $switch-height);
+      border: 1px solid #000;
```
Comment (Magnus-G) [inline source/stylesheets/refills/unstyled/_switch.scss]: I don't agree to this.

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        content: "";
+        width: $switch-width - ($switch-padding * 2);
+        z-index: 1;
+      }
+
+      &:after {
+        @include position(absolute, 1px 0 0 1px);
+        @include size($knob-size);
+        @include transition(all 0.3s ease);
+        background: #000;
```
Comment (Magnus-G) [inline source/stylesheets/refills/unstyled/_switch.scss]: I don't agree to this.

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        content: "";
+        z-index: 2;
+      }
+    }
+
+    &:checked {
+      + .checkbox {
+        background: #aaa;
```
Comment (Magnus-G) [inline source/stylesheets/refills/unstyled/_switch.scss]: I don't agree to this.

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        &:after {
```
Comment (Magnus-G) [inline source/stylesheets/refills/unstyled/_switch.scss]: In this case we need to...

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 5]
```diff
@@ -0,0 +1,50 @@
+.switch {  
+  $switch-width: 52px;
+  $switch-padding: 2px;
+  $switch-height: 32px;
+  $knob-size: $switch-height - ($switch-padding * 2);
+  $knob-width: $knob-size;
```
Comment (tysongach) [inline source/stylesheets/refills/unstyled/_switch.scss]: These variables should go above Line 1, `.switch`, outside the declaration block.

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      z-index: 0;
+
+      &:before {
+        @include position(absolute, $switch-padding 0 0 $switch-padding);
+        content: "";
+        width: $switch-width - ($switch-padding * 2);
+        z-index: 1;
+      }
+
+      &:after {
```
Comment (tysongach) [inline source/stylesheets/refills/unstyled/_switch.scss]: `:after` » `::after`

Ref: “[Use double colons for pseudo-elements](https://github.com/thoughtbot/guides/tree/master/style/sass#formatting)”

File: source/stylesheets/refills/unstyled/_switch.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        @include position(absolute, 1px 0 0 1px);
+        @include size($knob-size);
+        @include transition(all 0.3s ease);
```
Comment (tysongach) [inline source/stylesheets/refills/unstyled/_switch.scss]: No need to [prefix transitions anymore](http://caniuse.com/#feat=css-transitions).

File: source/stylesheets/refills/unstyled/_switch.scss (hunk 2 above)
Comment (tysongach) [inline source/stylesheets/refills/unstyled/_switch.scss]: `background` » `background-color`

File: source/stylesheets/refills/unstyled/_switch.scss (hunk 5 above)
Comment (Magnus-G) [inline source/stylesheets/refills/unstyled/_switch.scss]: Isn't there a risk in making the variables global? What if there's another `$switch-width` in the project? For some reason. The user might have created it somewhere else.

**Pull request closed without merging**